parser_library/src/processing/statement_analyzers/hit_count_analyzer.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2023 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "hit_count_analyzer.h"
+
+#include <cassert>
+#include <optional>
+
+#include "context/id_storage.h"
+#include "processing/error_statement.h"
+#include "processing/statement.h"
+
+namespace hlasm_plugin::parser_library::processing {
+
+hit_count_analyzer::hit_count_analyzer(context::hlasm_context& ctx)
+    : m_ctx(ctx)
+{}
+
+namespace {
+std::optional<range> get_range(const context::hlasm_statement* stmt,
+    statement_provider_kind prov_kind,
+    processing_kind proc_kind,
+    const context::hlasm_context& ctx)
+{
+    if (!stmt)
+        return std::nullopt;
+
+    const auto adj_range = [prov_kind, &proc_kind, &ctx](range r) -> std::optional<range> {
+        if (proc_kind != processing_kind::LOOKAHEAD && r.start == r.end)
+            return std::nullopt;
+
+        if (prov_kind == statement_provider_kind::OPEN)
+            r.end.line = ctx.current_source().end_index - 1;
+        r.end.column = 72;
+        return r;
+    };
+
+    if (stmt->kind == context::statement_kind::RESOLVED)
+    {
+        if (auto resolved = stmt->access_resolved(); resolved)
+            return adj_range(resolved->stmt_range_ref());
+    }
+    else if (stmt->kind == context::statement_kind::DEFERRED)
+    {
+        if (auto deferred = stmt->access_deferred(); deferred)
+            return adj_range(deferred->stmt_range_ref());
+    }
+
+    return std::nullopt;
+}
+
+size_t& emplace_hit_count(
+    hit_count_map& hc_map, const utils::resource::resource_location& rl, range r, bool can_be_external_macro)
+{
+    static constexpr auto map_emplacer = []<typename VALUE>(auto& m, const auto& k) {
+        if (auto m_it = m.find(k); m_it != m.end())
+            return m_it;
+
+        return m.try_emplace(k, VALUE {}).first;
+    };
```

Comment:
m.try_emplace(k) should work just fine, right? just add `size_t count = 0;` in hit_count_details